confluent_kafka/src/Producer.c

```diff
@@ -398,6 +398,22 @@ static PyObject *Producer_flush (Handle *self, PyObject *args,
         return cfl_PyInt_FromInt(qlen);
 }
 
+
+static PyObject *Producer_purge (Handle *self, PyObject *args,
+                                 PyObject *kwargs) {
+	int blocking = 0;
+        static char *kws[] = { "blocking", NULL };
+
+	if (!PyArg_ParseTupleAndKeywords(args, kwargs, "|p", kws, &blocking))
+			return NULL;
+	if (blocking==1)
+		rd_kafka_purge(rk, RD_KAFKA_PURGE_F_QUEUE)
```

Comment:
Make sure to handle any errors which may be raised from the execution of rd_kafka_purge 

https://github.com/edenhill/librdkafka/blob/master/src/rdkafka.h#L3478-L3482